src/components/homepage/EditorHeroSection/EditorHeroSection.jsx

```diff
@@ -0,0 +1,119 @@
+
+const HeroComponent = ({ heroType, fieldId }) => {
+  if (heroType === "highlights") return <HeroHighlight fieldId={fieldId} />
+  if (heroType === "dropdown")
+    return <HeroDropdown fieldId={`${fieldId}.dropdown`} />
+  return null
+}
+
+export const EditorHeroSection = ({
+  fieldId, //  This fieldId refers to sections.0.hero, and it's a string reference id to the object in useForm
+}) => {
+  const {
+    register,
+    unregister,
+    formState: { errors },
+    setValue,
+    watch,
+  } = useFormContext()
+  const sectionErrors = _.get(errors, fieldId)
+  const heroType = watch(`${fieldId}.heroType`)
+
+  useEffect(() => {
+    // reset highlight and dropdown toggle states
+    if (heroType !== "highlights") {
+      setValue(`${fieldId}.key_highlights`, [])
+      unregister(`${fieldId}.button`)
+      unregister(`${fieldId}.url`)
+    }
+    if (heroType !== "dropdown") {
+      unregister(`${fieldId}.dropdown`)
+    }
+  }, [heroType, unregister, setValue])
+
+  return (
+    <CardContainer cardTitle="Hero Section" isError={!_.isEmpty(sectionErrors)}>
+      <FormField
+        register={register}
+        title="Hero title"
+        id={`${fieldId}.title`}
+        errorMessage={sectionErrors?.title?.message}
+        isRequired
+      />
+      <FormField
+        register={register}
+        title="Hero subtitle"
+        id={`${fieldId}.subtitle`}
+        errorMessage={sectionErrors?.subtitle?.message}
+        isRequired
+      />
+      <FormContext
+        hasError={!!sectionErrors?.background?.message}
+        isRequired
+        onFieldChange={(e) => setValue(`${fieldId}.background`, e.target.value)}
+      >
+        <FormTitle>Hero background image</FormTitle>
+        <FormFieldMedia
+          id={`${fieldId}.background`}
+          inlineButtonText="Choose Image"
+          register={register}
+        />
+        <FormError>{sectionErrors?.background?.message}</FormError>
+      </FormContext>
+      <br />
+      <>
+        <p className={elementStyles.formLabel}>Hero Section Type</p>
+        {/* Permalink or File URL */}
+        <div className="d-flex flex-column">
+          <label className="flex-fill">
+            <input
+              type="radio"
+              {...register(`${fieldId}.heroType`)}
+              id="radio-highlights"
+              name={`${fieldId}.heroType`}
+              value="highlights"
+            />
+            Highlights + Button
+          </label>
+          <label htmlFor="radio-dropdown" className="flex-fill">
+            <input
+              type="radio"
+              {...register(`${fieldId}.heroType`)}
+              id="radio-dropdown"
+              name={`${fieldId}.heroType`}
+              value="dropdown"
+            />
+            Dropdown
+          </label>
+          <label htmlFor="radio-none" className="flex-fill">
+            <input
+              type="radio"
+              {...register(`${fieldId}.heroType`)}
```

Comment:
i think this might cause bugs down the line - RHF explicitly states that names must be unique (see [here](https://react-hook-form.com/api/useform/register), under rules)